common/src/com/thoughtworks/go/util/TimeConverter.java

```diff
@@ -127,6 +127,18 @@ private static SimpleDateFormat getDateFormatterWithTimeZone() {
         return new SimpleDateFormat("d MMM yyyy HH:mm 'GMT' Z", Locale.ENGLISH);
     }
 
+    public String getHumanReadableString(Date date) {
+
+        return date == null ? ConvertedTime.NOT_AVAILABLE.toString() : new SimpleDateFormat("dd MMM, yyyy 'at' HH:mm:ss", Locale.ENGLISH)
```

Comment:
Creating new instances of `SimpleDateFormat` is expensive. Formatting dates using that is expensive too. Can't make it static because it's not thread safe either.

Consider using a static instance of `org.joda.time.format.DateTimeFormatter` instead.